integration/e2ecortex/services.go

```diff
@@ -237,6 +237,9 @@ func NewSingleBinary(name string, flags map[string]string, image string, otherPo
 			"-ingester.concurrent-flushes":   "10",
 			"-ingester.max-transfer-retries": "10",
 			"-ingester.num-tokens":           "512",
+			// Ruler
+			"-ruler.storage.type":            "local",
+			"-ruler.storage.local.directory": "/tmp/cortex/rules",
```

Comment:
Where is this used? I think you override it when used. I'm wondering if we need to have this configuration at all in the generic `NewSingleBinary()` instead of injecting it when required (given in the integration test you override it anyway).

Reply:
The single binary now requires _some_ amount of ruler configuration or it will fail to startup.  This is a relatively minimal configuration that will allow the single binary to start.  Without this tests such as `TestSingleBinaryWithMemberlist` will fail.

We could make the Ruler startup successfully with no configuration using an in memory implementation if desired.  I'd be concerned with starting up in a configuration like this however as the Ruler probably wouldn't work as expected.

Reply:
I think we should mention the fact that minimal ruler configuration is required in the changelog. Alternatively, we can disable ruler if no configuration is provided in single-binary mode (and log warning about it).

As it is now, people using single-binary will run into this problem as soon as they try to run Cortex, with pretty cryptic error:

```
cortex-1_1       | level=error ts=2020-07-14T11:48:54.9596355Z caller=log.go:140 msg="error running cortex" err="configs_api_url is not set or valid\nerror initialising module: ruler\ngithub.com/cortexproject/cortex/pkg/util/modules.(*Manager).InitModuleServices\n\t/Users/peter/Grafana/cortex/pkg/util/modules/modules.go:89\ngithub.com/cortexproject/cortex/pkg/cortex.(*Cortex).Run\n\t/Users/peter/Grafana/cortex/pkg/cortex/cortex.go:299\nmain.main\n\t/Users/peter/Grafana/cortex/cmd/cortex/main.go:153\nruntime.main\n\t/usr/local/Cellar/go/1.14.3/libexec/src/runtime/proc.go:203\nruntime.goexit\n\t/usr/local/Cellar/go/1.14.3/libexec/src/runtime/asm_amd64.s:1373"
```

Reply:
Per Peter's comments I have removed the need for ruler configuration in the single binary.  Resolving.